#!/usr/bin/env python
PACKAGE = "aerial_robot_control"

from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels
from math import pi

gen = ParameterGenerator()

gen.add("pid_control_flag", bool_t, DynamicReconfigureLevels.RECONFIGURE_CONTROL_FLAG, "XY PID Control Flag", False)
gen.add("p_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_P_GAIN, "p gain", 0.01, 0, 50)
gen.add("i_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_I_GAIN, "i gain", 0.01, 0, 20)
gen.add("d_gain", double_t, DynamicReconfigureLevels.RECONFIGURE_D_GAIN, "d gain", 0.15, 0, 50)

exit(gen.generate(PACKAGE, "aerial_robot_control", "PID"))


